src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -381,15 +381,32 @@ def schedule_unit(self, unit):
         is_last  = False
         tag      = cud.get('tag')
 
+        # - PRRTE related - start -
+        dvm_hosts_list = self._rm_lm_info.get('dvm_hosts') or []
+        # use tag as a DVM ID, so task will be allocated to nodes that are
+        # handled by a corresponding DVM (only applied with `dvm_hosts_list`)
+        # FIXME: new coming attribute `tags` would consider `dvm_id`
+        if dvm_hosts_list and tag is not None and tag not in self._tag_history:
+            try:
+                tag = int(tag)
+            except (TypeError, ValueError):
+                tag = None
+            else:
+                if len(dvm_hosts_list) <= tag:
+                    raise ValueError('dvm_id (%s) out of range' % tag)
+                self._tag_history[tag] = dvm_hosts_list[tag]
+        unit_dvm_id = None
+        # - PRRTE related - end -
+
         # what remains to be allocated?  all of it right now.
         alc_slots = list()
         rem_slots = req_slots
 
         # start the search
         for node in self._iterate_nodes():
 
-            node_uid  = node['uid']
-          # node_name = node['name']
+          # node_uid  = node['uid']
+            node_name = node['name']
```

Comment:
Oh, why was that switched from `node_uid` to `node_name`?  Specifically for localhost and debug modes, node names are not necessarily unique, but the UIDs are.  Am I missing something?

Reply:
> Specifically for localhost and debug modes

yeah, I didn't think about that... then both values should be used - one for DVM and another for other cases. `node_name` is needed for DVM case, since `dvm_host_list` contains a list of host names

Reply:
Wouldn't it make sense to then use `node_id` in the host list, and to look up the `node_name` as needed?

Reply:
I'll update `dvm_host_list` then, so I'll keep `node_id`s there. Just at the beginning I've started to store host names there (names we put into `prrte.hosts` file), but since RP handles that then to keep ids is valid as well

Reply:
updated